internal/cmd/cmd_test.go

```diff
@@ -0,0 +1,48 @@
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+// Test if a CMD can start and stop their own server when the primary server isn't being run.
+// Since all CMDs use the same infrastructure, we only need to test it for one CMD.
+func TestCMDStartsServer(test *testing.T) {
+	defer config.UNIT_TESTING_MODE.Set(config.UNIT_TESTING_MODE.Get())
+	config.UNIT_TESTING_MODE.Set(true)
+
+	defer util.RemoveRecordedTempDirs()
+
+	expectedSubstrings := []struct {
+		expectedSubstring string
+	}{
+		{`API Server Started.`},
+		{`Unix Socket Server Started.`},
+		{`API Server Stopped.`},
+		{`Unix Socket Server Stopped.`},
+	}
```

Comment:
Would it be safe to assume then that if userlist comes back with the expected list of users, that means it started it's own server? If that's the case then it may not be necessary to have these expected substrings, because as you said, they can easily change often.